activerecord/test/cases/migration/columns_test.rb

```diff
@@ -368,6 +368,23 @@ def test_remove_columns_single_statement
       ensure
         connection.drop_table :my_table, if_exists: true
       end
+
+      def test_add_timestamps_single_statement
+        connection.create_table "my_table"
+
+        # SQLite3's ALTER TABLE statement has several limitations. To manage
+        # this, the adapter creates a temporary table, copies the data, drops
+        # the old table, creates the new table, then copies the data back.
+        expected_query_count = current_adapter?(:SQLite3Adapter) ? 12 : 1
```

Comment:
For further review (mainly for me), here are 12 SQL statements executed with SQLite3Adapter.
```sql
PRAGMA foreign_keys
PRAGMA defer_foreign_keys
PRAGMA defer_foreign_keys = ON
PRAGMA foreign_keys = OFF
CREATE TEMPORARY TABLE "amy_table" ("id" integer NOT NULL PRIMARY KEY)
INSERT INTO "amy_table" ("id") SELECT "id" FROM "my_table"
DROP TABLE "my_table"
CREATE TABLE "my_table" ("id" integer NOT NULL PRIMARY KEY, "created_at" datetime(6) NOT NULL, "updated_at" datetime(6) NOT NULL)
INSERT INTO "my_table" ("id") SELECT "id" FROM "amy_table"
DROP TABLE "amy_table"
PRAGMA defer_foreign_keys = 0
PRAGMA foreign_keys = 1
```